src/ucc/api/ucc.h

```diff
@@ -1391,12 +1391,22 @@ typedef enum {
                                                             Particularly, useful
                                                             for alltoallv
                                                             operation.*/
-    UCC_COLL_ARGS_FLAG_CONTIG_DST_BUFFER    = UCC_BIT(5)  /*!<If set, the dst
+    UCC_COLL_ARGS_FLAG_CONTIG_DST_BUFFER    = UCC_BIT(5), /*!<If set, the dst
                                                             buffer is considered
                                                             contiguous.
                                                             Particularly, useful
                                                             for alltoallv
                                                             operation. */
+    UCC_COLL_ARGS_FLAG_TIMEOUT              = UCC_BIT(6)  /*!<If set, the collective
+                                                            elapsed time is being measured
+                                                            and compared vs the provided
+                                                            timeout value. UCC_ERR_TIMED_OUT
+                                                            will be returned from collective
+                                                            test if elapsed time exceeds
+                                                            the timeout.
+                                                            Note, the status is not guaranteed
+                                                            to be global on all the processes
+                                                            participating in the collective*/
```

Comment:
I'm guessing you are saying that the timeout applies only to the calling thread.

Reply:
yes, "reported" (via collective_test) timeout is valid only the calling thread. How should i better formulate that?

Reply:
@vspetrov I'm not sure, if this better :) `"If set and the elapsed time after call invocation is greater than @ref ucc_coll_args_t.timeout, the library returns UCC_ERR_TIMED_OUT on the calling thread. There is no impact on threads or processes that does not set the value. "`